packages/selfcare-client-users-updater/src/services/selfcareClientUsersUpdaterProcessor.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  getAllFromPaginated,
+  getInteropHeaders,
+  logger,
+  RefreshableInteropToken,
+  userRole,
+} from "pagopa-interop-commons";
+import { EachMessagePayload } from "kafkajs";
+import {
+  generateId,
+  CorrelationId,
+  genericInternalError,
+} from "pagopa-interop-models";
+import { bffApi } from "pagopa-interop-api-clients";
+import { AuthorizationProcessClient } from "../clients/authorizationProcessClient.js";
+import {
+  selfcareUserEventType,
+  UsersEventPayload,
+  relationshipStatus,
+} from "../model/UsersEventPayload.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function selfcareClientUsersUpdaterProcessorBuilder(
+  refreshableToken: RefreshableInteropToken,
+  authorizationProcessClient: AuthorizationProcessClient,
+  productId: string
+) {
+  return {
+    async processMessage({
+      message,
+      partition,
+    }: EachMessagePayload): Promise<void> {
+      const correlationId: CorrelationId = generateId();
+
+      const loggerInstance = logger({
+        serviceName: "selfcare-client-users-updater",
+        correlationId,
+      });
+
+      try {
+        loggerInstance.info(
+          `Consuming message for partition ${partition} with offset ${message.offset}`
+        );
+
+        if (!message.value) {
+          loggerInstance.warn(
+            `Empty message for partition ${partition} with offset ${message.offset}`
+          );
+          return;
+        }
+
+        const stringPayload = message.value.toString();
+        const jsonPayload = JSON.parse(stringPayload);
+
+        if (jsonPayload.productId !== productId) {
+          loggerInstance.info(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Not required product: ${jsonPayload.productId}`
+          );
+          return;
+        }
+
+        const userEventPayload = UsersEventPayload.parse(jsonPayload);
+        const eventUserId = userEventPayload.user.userId;
+        if (!eventUserId) {
+          loggerInstance.warn(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Missing userId.`
+          );
+          return;
+        }
+
+        if (
+          userEventPayload.eventType === selfcareUserEventType.update &&
+          userEventPayload.user.productRole === userRole.ADMIN_ROLE &&
+          userEventPayload.user.relationshipStatus !== relationshipStatus.active
+        ) {
```

Comment:
We could also handle these conditions with ts-pattern

```suggestion
match([eventUserId, userEventPayload])
    .with([P.nullish, P._], () => {
      loggerInstance.warn(
        `Skipping message for partition ${partition} with offset ${message.offset} - Missing userId.`
      );
      return;
    })
    .with(
      [
        P.string,
        {
          eventType: P.when(type => type === selfcareUserEventType.update),
          user: {
            productRole: P.when(role => role === userRole.ADMIN_ROLE),
            relationshipStatus: P.not(relationshipStatus.active)
          }
        }
      ],
      async () => {
```
and also handle the last `else` with `.otherwise`